src/vs/workbench/services/textfile/common/textFileEditorModel.ts

```diff
@@ -334,6 +356,9 @@ export class TextFileEditorModel extends BaseTextEditorModel implements ITextFil
 				diag('makeDirty() - prevented save because we are in conflict resolution mode', this.resource, new Date());
 			}
 		}
+
+		// Trigger listeners only if no auto save occurred and model changes were not blocked
+		this._onDidContentChange.fire();
```

Comment:
@Tyriar I was more thinking about the event in general: if the model content changes, the event should fire. Looking at the code it seems there are return statements before this event is fired and thus a listener would not get an event if the user e.g. undos to the point of the saved state. I would (for performance reasons) not like to introduce a real event with state here.